src/core/dom.js

```diff
@@ -105,6 +105,27 @@ export function removeNodes(node, callbackFn) {
   }
 }
 
+/**
+ * Remove all node’s child nodes that pass the test regardless of their depth.
+ * @param {Node} node the node to process
+ * @param {Function} callbackFn the predicate
+ */
+export function removeAllNodes(node, conditionFn) {
+  // Remove the node if it meets the condition
+  if (conditionFn(node)) {
+    node.remove();
+    return;
+  }
+
+  // Loop through the node’s children
+  for (const el of [...node.childNodes]) {
+    if (el.nodeType === Node.ELEMENT_NODE) {
```

Comment:
On peut passer directement par la propriété `node.children` pour faire ça
https://developer.mozilla.org/en-US/docs/Web/API/Element/children

Mais sinon je ne sais pas si il faut nécessairement filtrer les éléments sur Node.ELEMENT_NODE dans le for

Reply:
Ah oui ! Mais tu préfères que je laisse juste `[...node.childNodes]` sans filtrer ?

Reply:
ça correspond mieux au nom et à la description.
D'ailleurs `removeAllNodes` -> `removeNodesRecursively`
`Remove all node’s child nodes that pass the test regardless of their depth.` -> `Remove recursively all node's child nodes that pass the test implemented by the provided function.`

Reply:
D’accord ! 
Mais du coup je me demande si je dois laisser cette condition `el.nodeType === Node.ELEMENT_NODE` à l'utilisation de la fonction ici : https://github.com/lesjoursfr/edith/blob/remove-empty-tags/src/ui/editor.js#L106